facade/servicetemplate.go

```diff
@@ -446,3 +437,26 @@ func reloadLogstashContainerImpl(ctx datastore.Context, f FacadeInterface) error
 	}
 	return nil
 }
+
+func pullServiceImages(svcDef *servicedefinition.ServiceDefinition) error {
+	return pullImages(getImageIDs(*svcDef))
+}
+
+func pullImages(imgs []string) error {
+	for _, img := range imgs {
+		imageID, err := commons.ParseImageID(img)
+		if err != nil {
+			return err
+		}
+		tag := imageID.Tag
+		if tag == "" {
+			tag = "latest"
+		}
+		image := fmt.Sprintf("%s:%s", imageID.BaseName(), tag)
+		glog.Infof("Pulling image %s", image)
+		if err := docker.PullImage(image); err != nil {
+			glog.Warningf("Unable to pull image %s", image)
```

Comment:
Are you talking about the warning in 458? If so, that was in the original code (I just extracted it to another method to avoid duplication). I think they used a warning there so that we would pull as many docker images as possible, even if one failed.